src/MALA.jl

```diff
@@ -69,10 +72,16 @@ function AbstractMCMC.step(
 
     # Decide whether to return the previous params or the new one.
     transition = if -Random.randexp(rng) < logα
-        GradientTransition(candidate, logdensity_candidate, gradient_logdensity_candidate)
+        accepted_draws = sampler.accepted_draws + 1
+        GradientTransition(candidate, logdensity_candidate, gradient_logdensity_candidate, true)
     else
-        transition_prev
+        accepted_draws = sampler.accepted_draws
+        candidate = transition_prev.params
+        lp = transition_prev.lp
+        gradient = transition_prev.gradient
+        GradientTransition(candidate, lp, gradient, false)
     end
+    sampler = MALA(sampler.proposal, accepted_draws, sampler.total_draws+1)
```

Comment:
This won't work AFAICT, the sampler object is not stored (and IMO should not be used to store these properties anyway, see my comment above).